doc/source/whatsnew/v1.1.0.rst

```diff
@@ -293,6 +293,9 @@ Other enhancements
 - :meth:`~pandas.io.json.read_json` now accepts `nrows` parameter. (:issue:`33916`).
 - :meth `~pandas.io.gbq.read_gbq` now allows to disable progress bar (:issue:`33360`).
 - :meth:`~pandas.io.gbq.read_gbq` now supports the ``max_results`` kwarg from ``pandas-gbq`` (:issue:`34639`).
+- :func:`pandas.read_csv` now accepts engine="pyarrow" as an argument, allowing for faster csv parsing
+  if pyarrow>=0.15 is installed. However, the pyarrow engine is less feature-complete than its "c" or
+  "python" counterparts. See the :doc:`I/O docs </user_guide/io>` for more info. (:issue:`23697`)
```

Comment:
I think we could actually create a dedicated subsection in the "Enhancements" section for this, as this is a nice new feature we should highlight (it basically enables (at least for simple csv's) to read in parallel, and thus can give a significant speed-up)